botkit.d.ts

```diff
@@ -0,0 +1,427 @@
+            open: ApiMethod;
+            rename: ApiMethod;
+            setPurpose: ApiMethod;
+            setTopic: ApiMethod;
+            unarchive: ApiMethod;
+        };
+        im: {
+            close: ApiMethod;
+            history: ApiMethod;
+            list: ApiMethod;
+            mark: ApiMethod;
+            open: ApiMethod;
+        };
+        mpim: {
+            close: ApiMethod;
+            history: ApiMethod;
+            list: ApiMethod;
+            mark: ApiMethod;
+            open: ApiMethod;
+        };
+        pins: {
+            add: ApiMethod;
+            list: ApiMethod;
+            remove: ApiMethod;
+        };
+        reactions: {
+            add: ApiMethod;
+            get: ApiMethod;
+            list: ApiMethod;
+            remove: ApiMethod;
+        };
+        rtm: {
+            start: ApiMethod;
+        };
+        search: {
+            all: ApiMethod;
+            files: ApiMethod;
+            messages: ApiMethod;
+        };
+        stars: {
+            add: ApiMethod;
+            list: ApiMethod;
+            remove: ApiMethod;
+        };
+        team: {
+            accessLogs: ApiMethod;
+            info: ApiMethod;
+        };
+        users: {
+            getPresence: ApiMethod;
+            info: ApiMethod;
+            list: ApiMethod;
+            setActive: ApiMethod;
+        };
+        callAPI(command: string, data: any, cb: (err: string | Error, response: any) => void): never;
+        callAPIWithoutToken(command: string, data: any, cb: (err: string | Error, response: any) => void): never;
+    }
+    class Bot {
+        type: string;
+        botkit: Controller;
+        config: Configuration;
+        utterances: Utterances;
+        api: WebAPI;
+        identity: {
+            id: string;
+            name: string;
+        };
+
+        constructor(controller: Controller, config: Configuration);
+        configureIncomingWebhook(options: { url: string }): Bot;
+        sendWebhook(options, cb): any;
+        configureRTM(config): Bot;
+        closeRTM(err: Error): never;
+        reconnect(err: Error): any;
+        destroy(): never;
+        startRTM(cb: (err: string | Error, bot: Bot, res) => void): Bot;
+        identifyBot(cb: (err: string | Error, data) => void): any;
+        identifyTeam(cb: (err: string | Error, id: string) => void): string;
+        startPrivateConversation(message: Message, cb: (err: Error, convo: Conversation) => void): never;
+        startConversation(message: Message, cb: (err: Error, convo: Conversation) => void): never;
+        createConversation(message: Message, cb: (err: Error, convo: Conversation) => void): never;
+        private _startDM(task: Task, user_id: string, cb: Function): any;
+        send(message: Message, cb: (err?: string | Error) => void): never;
+        replyAcknowledge(cb: (err?: string | Error) => void): never;
+        replyPublic(src: any, resp: string, cb: (err?: string | Error) => void): never;
```

Comment:
All of the reply methods also support a different reply type:
```
    interface Reply {
        username: string,
        attachments: Array<Attachment>
    }
```
`resp: string` should be `resp: string | Reply` for each of the following:

- replyPublic
- replyPublicDelayed
- replyPrivate
- replyPrivateDelayed
- replyInteractive
- reply
- replyWithTyping
- replyAndUpdate